localstack-core/localstack/services/events/utils.py

```diff
@@ -183,3 +208,31 @@ def re_format_event(event: FormattedEvent, event_bus_name: EventBusName) -> PutE
     if event.get("replay-name"):
         re_formatted_event["ReplayName"] = event["replay_name"]
     return re_formatted_event
+
+
+def get_trace_header_encoded_region_account(
+    event: PutEventsRequestEntry | FormattedEvent | TransformedEvent,
+    source_region: str,
+    source_account_id: str,
+    target_region: str,
+    target_account_id: str,
+) -> str | None:
+    """Encode the original region and account_id for cross-region and cross-account
```

Comment:
That trace header encoding is a hack we likely need to revisit once we add X-Ray support. It might be worth clarifying that it's a hack and not expected behavior (currently unclear from the comment).